### PR TITLE
chore(client): remove extraneous generics

### PR DESCRIPTION
**Pull request by harbu**
## Summary

Remove extraneous and occasionally misleading generics. Many were related to`StreamMessage<T>` where the type `T` could not actually  be guaranteed to be what is claimed.

## Limitations and future improvements

- Potentially some missed spots here and there. However, public API should be pretty clean from these now.

## Checklist before requesting a review

- [ ] Is this a breaking change? If it is, be clear in summary.
- [x] Read through code myself one more time.
- [x] Make sure any and all `TODO` comments left behind are meant to be left in.
- [x] Has reasonable passing test coverage?
- [ ] Updated changelog if applicable.
- [x] Updated documentation if applicable.

### CODE REVIEW

File: packages/client/src/StreamrClient.ts [hunk 1]
```diff
@@ -102,13 +102,13 @@ export class StreamrClient {
      * @category Important
      *
      * @param streamDefinition - the stream or stream partition to publish the message to
-     * @param content - the content (the payload) of the message
+     * @param content - the content (the payload) of the message (usually an JSON serializable object)
```
Comment (teogeb) [inline packages/client/src/StreamrClient.ts]: Needs to be JSON serializable object as we create a message which has content type of JSON (the default content type, `StreamMessage:107`)

File: packages/client/src/subscribe/OrderMessages.ts [hunk 2]
```diff
@@ -146,8 +146,8 @@ export class OrderMessages<T> {
         }
     }
 
-    transform(): (src: AsyncGenerator<StreamMessage<T>, any, unknown>) => AsyncGenerator<StreamMessage<T>, void, unknown> {
-        return async function* Transform(this: OrderMessages<T>, src: AsyncGenerator<StreamMessage<T>>) {
+    transform(): (src: AsyncGenerator<StreamMessage, any, unknown>) => AsyncGenerator<StreamMessage, void, unknown> {
```
Comment (teogeb) [inline packages/client/src/subscribe/OrderMessages.ts]: Could simplify AsyncGenerator generics to be just `AsyncGenerator<StreamMessage>` (in this PR on in a separate PR)

File: packages/client/src/StreamrClient.ts (hunk 1 above)
Comment (harbu) [reply packages/client/src/StreamrClient.ts]: Addressed in a5e9c88e854ce597f323272888a637a5150aa834 with wording change "must be"

File: packages/client/src/subscribe/OrderMessages.ts (hunk 2 above)
Comment (harbu) [reply packages/client/src/subscribe/OrderMessages.ts]: Fixed in a5e9c88e854ce597f323272888a637a5150aa834

**Pull request merged**